### DISCUSSION TITLE
cph.plot_partial_effects_on_outcome()

### DISCUSSION THREAD

**Discussion by alexandre-sousa-git** [CamDavidsonPilon/lifelines#1299] [General]
Hello ,

I'm new with this library , and i have a question about this function, is it possible extract the values plotted by the function?

**Comment by CamDavidsonPilon** [comment]
Hi @alexandre-sousa-git - no, there isn't, but this has been brought up before. Are you interested in the the baseline survival line too?

**Reply by alexandre-sousa-git** [reply]
Thank you very much for the quick response.
To the baseline survival line is the function : "baseline_survival_" right?

**Reply by CamDavidsonPilon** [reply]
That's right

**Reply by alexandre-sousa-git** [reply]
What I really needed was to know the coordinates of each curve of each covariate not the baseline. For example the difference of the survival for each gender .

**Reply by CamDavidsonPilon** [reply]
I mean, two ways to do that: 
1. Look at the regression coefficient on gender with `.print_summary()` - that provides a summary of the survival differences
2. Manually determine the survival curves:
```python
cph = CoxPHFitter().fit(rossi, "week", "arrest")

baseline = cph._central_values.copy()
cph.predict_survival_function(baseline) # baseline

X1 = baseline.copy()
X1['baseline', 'arrest'] = 1
cph.predict_survival_function(X1) # alternative

```

**Reply by alexandre-sousa-git** [reply]
Thanks a lot! This will help! congrats for your work

**Reply by dlovell** [reply]
@CamDavidsonPilon : I believe this snippet will extract the data from the `AxesSubplot` returned by `plot_partial_effects_on_outcome`

```
pd.DataFrame({
    line.get_label(): pd.Series(
        line.get_ydata(),
        line.get_xdata(),
    )
    for line in ax.lines
})
```
